### PR TITLE
sink(ticdc): add more log and metrics to storage sink

### LINKED ISSUE #10115: add more log and metrics to storage sink
In order to better monitor the running status of storage sink, we need to add some logs and monitoring to related modules.

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: close #10115

### What is changed and how it works?
 add more log and metrics to storage sink, the write duration panel will take effect after #9954 is merged.
image

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

If you don't think this PR needs a release note then fill it with `None`.
```

### CODE REVIEW

File: pkg/config/sink.go [hunk 1]
```diff
@@ -221,7 +221,8 @@ func (c *CSVConfig) validateAndAdjust() error {
 		}
 	default:
 		return cerror.WrapError(cerror.ErrSinkInvalidConfig,
-			errors.New("csv config delimiter contains more than one character"))
+			errors.New("csv config delimiter contains more than one character, note that escape "+
```
Comment (3AceShowHand) [inline pkg/config/sink.go:224]: I think you can use `GenWithStack`.

Comment (CharlesCheung96) [reply pkg/config/sink.go:224]: This function is called internally by WrapError, but WrapError is safer in the case of nil errors.

**Review by asddongmen — APPROVED** [review]

**Review by 3AceShowHand — APPROVED** [review]

**Pull request merged**